sdk/ai/Azure.AI.Projects/samples/Sample24_Agent_Connected_Agent.md

```diff
@@ -0,0 +1,199 @@
+# Sample for use of an agent with a connected agent in Azure.AI.Projects.
+
+To enable your Agent to use a connected agent, you use `ConnectedAgentToolDefinition` along with the agent id, name, and a description.
+1. First we need to create agent client and read the environment variables, which will be used in the next steps.
+
+```C# Snippet:ConnectedAgent_CreateProject
+var connectionString = System.Environment.GetEnvironmentVariable("PROJECT_CONNECTION_STRING");
+var modelDeploymentName = System.Environment.GetEnvironmentVariable("MODEL_DEPLOYMENT_NAME");
+
+var projectClient = new AIProjectClient(connectionString, new DefaultAzureCredential());
+
+AgentsClient agentClient = projectClient.GetAgentsClient();
+```
+
+2. Next we will create the connected agent using the agent client. This agent will be used to initialize the `ConnectedAgentToolDefinition`.
+
+Synchronous sample:
+```C# Snippet:ConnectedAgent_CreateConnectedAgent
+Agent connectedAgent = agentClient.CreateAgent(
+   model: modelDeploymentName,
+   name: "stock_price_bot",
+   instructions: "Your job is to get the stock price of a company. If you don't know the realtime stock price, return the last known stock price.");
+
+ConnectedAgentToolDefinition connectedAgentDefinition = new(new ConnectedAgentDetails(connectedAgent.Id, connectedAgent.Name, "Gets the stock price of a company"));
+```
+
+Asynchronous sample:
+```C# Snippet:ConnectedAgentAsync_CreateConnectedAgent
+Agent connectedAgent = await agentClient.CreateAgentAsync(
+   model: modelDeploymentName,
+   name: "stock_price_bot",
+   instructions: "Your job is to get the stock price of a company. If you don't know the realtime stock price, return the last known stock price.");
+
+ConnectedAgentToolDefinition connectedAgentDefinition = new(new ConnectedAgentDetails(connectedAgent.Id, connectedAgent.Name, "Gets the stock price of a company"));
+```
+
+3. We will use the `ConnectedAgentToolDefinition` during the agent initialization.
+
+Synchronous sample:
+```C# Snippet:ConnectedAgent_CreateAgent
+Agent agent = agentClient.CreateAgent(
+   model: modelDeploymentName,
+   name: "my-assistant",
+   instructions: "You are a helpful assistant, and use the connected agent to get stock prices.",
+   tools: [ connectedAgentDefinition ]);
+```
+
+Asynchronous sample:
+```C# Snippet:ConnectedAgentAsync_CreateAgent
+Agent agent = await agentClient.CreateAgentAsync(
+   model: modelDeploymentName,
+   name: "my-assistant",
+   instructions: "You are a helpful assistant, and use the connected agent to get stock prices.",
+   tools: [ connectedAgentDefinition ]);
+```
+
+4. Now we will create the thread, add the message , containing a question for agent and start the run.
```

Comment:
NIT: Remove space in "message ," (but I think Markdown does not render it any way)